### ISSUE TITLE
error: expected value in expression #if !defined(HAVE_UNISTD_H) || HAVE_UNISTD_H

### ISSUE THREAD

**Issue by sambthompson** [c-ares/c-ares#787]
Builds of 1.30.0 broken on newer [macOS 10.13](https://build.macports.org/builders/ports-10.13_x86_64-builder/builds/237765/steps/install-port/logs/stdio) and [macOS 10.14](https://build.macports.org/builders/ports-10.14_x86_64-builder/builds/211988/steps/install-port/logs/stdio), as there is now a compile error on `ares_event_configchg.c`:

```
...
In file included from /opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_event_configchg.c:261:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/notify.h:35:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/dispatch/dispatch.h:36:45: error: expected value in expression
#if !defined(HAVE_UNISTD_H) || HAVE_UNISTD_H
                                            ^
1 error generated.

...
```

These errors not resolved/affected by patch to main MacPorts is carrying from c-ares 1.30 to main branch to land fixes for #782/#783.

**Comment by ryandesign** [comment]
Here is the same error when it was encountered by FreeRDP back when macOS 10.13 was released:

https://github.com/FreeRDP/FreeRDP/issues/4160

Their fix is not general, however, because they refer to the macro `__IOS__` which they define elsewhere in their build system; this is not a standard macro and it is wrong of them to define a macro with a name starting with double underscores since that is reserved for the implementation.

**Comment by bradh352** [comment]
What build system are you using? autotools or cmake?

**Comment by bradh352** [comment]
I'm assuming this is cmake, in which case 5804b30 would fix it.

**Comment by sambthompson** [comment]
> I'm assuming this is cmake, in which case [5804b30](https://github.com/c-ares/c-ares/commit/5804b30a878749865a7616bae49c4f5957aa8c0a) would fix it.

Thanks for this; yes, I believe it is cmake since using MacPorts. Will test ASAP; will have to spin up 10.14 machine.

**Comment by sambthompson** [comment]
OK, build test with this fix successful on 10.14. I hope it would work just as well on 10.13, but can't test this.

**Comment by bradh352** [comment]
c-ares 1.31.0 is now out which contains the legacy mac fixes
